code/datums/late_choices.dm

```diff
@@ -0,0 +1,90 @@
+// This datum holds the late choices UI for a new player
+// It should exist only as long as the menu is open
+/datum/late_choices
+	var/datum/vueui/ui
+	var/mob/abstract/new_player/NP
+
+/datum/late_choices/CanUseTopic(var/mob/user, var/datum/topic_state/state = default_state) // this is needed because VueUI closes otherwise
+	if(isnewplayer(user))
+		return STATUS_INTERACTIVE
+	return ..()
+
+/datum/late_choices/Topic(href, href_list)
+	// proxy Topic calls back to the user
+	NP.Topic(href, href_list)
+
+/datum/late_choices/New(var/mob/abstract/new_player/NP)
+	if(!istype(NP))
+		return
+	src.NP = NP
+	ui_open(NP)
+
+/datum/late_choices/proc/ui_open(mob/user)
+	ui = new(user, src, "late-choices", 330, 720, "Late-Join Choices")
+	ui.header = "minimal"
+	var/mob/mannequin = user.client.prefs.update_mannequin()
+	ui.add_asset("character", getFlatIcon(mannequin, SOUTH))
+	ui.open()
+
+/datum/late_choices/vueui_data_change(var/list/data, var/mob/user, var/datum/vueui/ui)
+	. = ..()
+	data = . || data || list()
+
+	VUEUI_SET_CHECK(data["round_duration"], get_round_duration_formatted(), ., data)
+	VUEUI_SET_CHECK(data["alert_level"], capitalize(get_security_level()), ., data)
+	VUEUI_SET_CHECK_IFNOTSET(data["character_name"], user.client.prefs.real_name, ., data)
+
+	var/shuttle_status = ""
+	if(emergency_shuttle) //In case Nanotrasen decides to reposess CentComm's shuttles.
+		if(emergency_shuttle.going_to_centcom()) //Shuttle is going to centcomm, not recalled
+			shuttle_status = "post-evac"
+		if(emergency_shuttle.online())
+			if (emergency_shuttle.evac) // Emergency shuttle is past the point of no recall
+				shuttle_status = "evac"
+			else // Crew transfer initiated
+				shuttle_status = "transfer"
+	VUEUI_SET_CHECK(data["shuttle_status"], shuttle_status, ., data)
+	VUEUI_SET_CHECK(data["alert_level"], capitalize(get_security_level()), ., data)
+
+	var/unique_role_available = FALSE
+	for(var/ghost_role in SSghostroles.spawners)
+		var/datum/ghostspawner/G = SSghostroles.spawners[ghost_role]
+		if(!G.show_on_job_select)
+			continue
+		if(G.cant_see(NP))
+			continue
+		unique_role_available = TRUE
+		break
+
+	VUEUI_SET_CHECK(data["unique_role_available"], unique_role_available, ., data)
+
+	var/jobs_available = 0
+	var/list/list/datum/job/jobs_by_department = DEPARTMENTS_LIST_INIT
+	for(var/datum/job/job in SSjobs.occupations)
+		if(job && NP.IsJobAvailable(job.title))
```

Comment:
```suggestion
		if(NP.IsJobAvailable(job.title))
```

Job should always exist here due to the filter in the for loop.

Although... SSjobs.occupations should always contain only /datum/job so...

```suggestion
for(var/J in SSjobs.occupations)
		var/datum/job/job = J
		if(job && NP.IsJobAvailable(job.title))
```

Maybe that is more appropriate?

Reply:
I think I'll go with the first variant, it's more ... "familiar"? You don't have to rely on the type of the list which is in a different file somewhere.